plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConfig.java

```diff
@@ -76,6 +76,13 @@
     private boolean ignorePublishAddress;
     private boolean verifyHostnames = true;
 
+    private boolean unionSchemaIndicesForAlias;
+    private boolean failOnAliasSchemaMismatch = true;
+    private String defaultDataTypeForAliasSchemaMismatch = "text";
```

Comment:
@findepi any thoughts here?

Reply:
Just noting that using keyword will cause pushdown to happen which may lead to incorrect results if the underlying type is compared differently in Trino vs ES. (e.g. Text equality in ES takes normalisation into account, keyword requires exact matches).

I feel text is the safer choice.

Reply:
Done

Reply:
@martint Here are my observations in ES and in Trino - https://github.com/trinodb/trino/pull/8202#discussion_r650354086.
Since ES is a NoSQL datastore, it doesn't validate the datatype. But in SQL we have to.

Reply:
So if we had a field that is represented as a `keyword` in one document, and a field that is a`long` in another document, both would be cast to `text`, it would be okay to project those values, and filtering would just cause an exception which I think works fine. It doesn't perfectly match ES behavior because the es behavior doesn't take types into account so nothing we can do here but make sure it is correct.

```
SELECT * 
FROM (VALUES (1, '3'), (1, '6'), (1, '7'), (1, '8')) t(a, b)
WHERE t.b = 5;
```

```
 Cannot apply operator: varchar(1) = integer
```